### ISSUE TITLE
Public Announcement - DO NOT UPGRADE ON MacOS SEQUOIA / 15.0 

### ISSUE THREAD

**Issue by lgandecki** [lwouis/alt-tab-macos#3672]
The version 6 has worked for me for a long time, I have no additional needs, it's configured, it works.

For whatever silly reason I was tempted to click the update to version 7 when it prompted me to do so. It stopped working. It doesn't start up when I try to open it using raycast/spotlight. When I open it from the terminal, it opens but it says it doesn't have permissions for screen recording. The permissions are there. I removed the existing ones, added it back, restart, same (I did it in many different combinations and order)

I wasted 15 minutes of my life and office time on this for really no reason :) But this is an essential piece of software, so I had to deal with it...

It's probably related to new permissions system or whatever, no care - old version works. If you did the mistake of clicking upgrade, you can download the old one from here: 
https://github.com/lwouis/alt-tab-macos/releases/tag/v6.73.0

remove the new one from applications first.

**Comment by lgandecki** [comment]
Oh, also, I'm sure it works for some of you guys - no care. Even if there is 5% risk it will break, why risk it? As far as I know there is no reason to upgrade alt-tab at all if things are working fine for you now.

**Comment by lwouis** [comment]
Duplicate of https://github.com/lwouis/alt-tab-macos/issues/3653

@lgandecki of course, if you don't want the new feature, remaining on the old version is great. v7.0.0 introduces a lot of new features.

**Comment by lgandecki** [comment]
well, right, but I think there should be a significant warning that it might simply stop working after an upgrade. I can't remember the last time I uprgaded something and it just broke completely in such an annoying way. 
I'm grateful for the app, it's fantastic! But honestly if this has such a high chance of failure, the app shouldn't prompt to an update. This is a piece of software that I, and I'm sure many people, rely on as a tiny but important tool to basically make a living.. :)

**Comment by lgandecki** [comment]
There is so many people in that thread talking about the issue, it's not "few people". Most people won't go to github to complain. I think it's honestly fair to warn people before updating, since there is no straightforward way to downgrade. Again, not everyone is very technical and knows their way around github to find release page etc.

Would it hurt to say in the update notes: "Note: for some number of users the app completely brakes after an update? We are working on a fix, meanwhile, don't update on MacOS 15.0"?

**Comment by lwouis** [comment]
I'm not sure to follow. We can't add such warning in advance. We don't know which bugs will happen.

We follow crashes using AppCenter. There are issues with every single version. Sometimes few, sometimes more. You haven't been impacted by them, so you think the latest version was special, but for another user, the previous version was the one that broke their setup.

We can't add a warning on every single update. Updating is always dangerous. This month, employees at my company update macOS, and all sorts of things broke for most users. Apple has the most money and the best talent, and like everyone else, they can't produce bug-free software.

Please be understanding of this. It's trully an impossible ask. Bugs are constantly there, in every version. They impact different users. We try to minimize their impact, and think about fallback scenarios and workaround. We do our best already 🙇

**Comment by lgandecki** [comment]
you can't add in advance, but if you have a known bug, people are at older version, you can release a patch, with a note warning about updating..

And yes system update that crash are super annoying but honestly I wouldn't click a MacOS update randomly at the start of my work day. But with a tiny app like this one.. I don't expect it to just don't open. Looking through the thread it seems there are people with many different issues.

I've been building software for living, and also spent countless hours on open source projects, often creating features and fixing bugs that were not relevant to me. I can't imagine a situation where I know that I introduced a situation where my app simply doesn't start after an upgrade for many people, and leave it at that, because "updating is dangerous".. 

I don't think I presume I'm special, did you have threads like that one previously? 

Anyway, we both spent enough time on this discussion. Good luck, and once again, thanks for the work. I followed on patreon (honestly I didn't even know this was a free open source software till it broke, sometimes it's hard to follow what I paid for 5 years ago and just keep using, what comes from setapp, and what's nicely packages OS app)
